### PR TITLE
Update compatibility for Django 4.0

### LINKED ISSUE #187: Support Django 4.0
`force_text` got removed from `django.utils.encoding` as used here: https://github.com/jazzband/django-downloadview/blob/0578729a66e0843bd34cf9253c81076f3b976408/django_downloadview/io.py#L4

See the release notes: https://docs.djangoproject.com/en/4.0/releases/4.0/#features-removed-in-4-0

### PR DESCRIPTION
**Pull request by tari**
The only documented compatibility change is removing use of `force_text` (which was
deprecated in Django 3.0) in favor of `force_str` (which has existed
since before Django 1.11). There are also some changes in middleware because Django's MiddlewareMixin started requiring a `get_response` argument to the constructor.

Fixes #187.

### CODE REVIEW

**Comment by Archmonger** [comment]
@tari Please add Django 4.0 as a [CI test case](https://github.com/jazzband/django-downloadview/blob/master/.github/workflows/test.yml).

**Comment by tari** [comment]
Added (as well as Python 3.10), but the tox configuration seems to duplicate parts of the test matrix and cause spurious failures; there seem to a few real failures as well that I have yet to investigate.

**Comment by Archmonger** [comment]
Only Django 3.2+ should support Python 3.10, so exclusions can be made for the rest.

File: .github/workflows/test.yml
Comment (Archmonger) [inline .github/workflows/test.yml:15]: duplicate comment

**Comment by tari** [comment]
I fixed the middleware issues in c286c0e0b01f0ffe5b03123d7134543895a3cd0f, but it's a little bit of a hack since it's just hiding the problem of crating a middleware without any `get_response` function; doing that became an error rather than simply deprecated in Django 4.0.

Breaking the "try each delegate middleware" logic out of `DownloadDispatcherMiddleware` so it can be used without creating something that looks like a middleware but isn't might be a nicer solution, but I opted for the more expedient hack for now since that also seems to be part of the public API (though I didn't see mention of it in the documentation)

**Comment by Archmonger** [comment]
I'm personally not the project lead for this repo, but since they haven't voiced on this PR I'll provide my input.

Migrating away from Nose can be done in a separate PR, but tests need to be passed on this PR before merging. Which means this PR can focus on Django 4.0 compatibility but exclude Python 3.10. Then a separate issue and PR should be created for 3.10 compatibility (which requires replacing nose).

However, my opinion is getting a proper non-hacky solution for Django 4.0 should be encompassed in this PR.

**Comment by Natim** [comment]
> this PR can focus on Django 4.0 compatibility but exclude Python 3.10

Agreed

**Comment by tari** [comment]
Understood- I'll work out a better solution to the middlewares, and leave Python 3.10 out of the mix for now.

**Comment by tari** [comment]
I've rebased this on top of #189 for Python 3.10 support, and made some more considered changes to the middleware that I'm happy with now.

File: .github/workflows/test.yml
Comment (Archmonger) [inline .github/workflows/test.yml:11]: Might want to bump max-parallel to `10` or more here since the matrix size has increased significantly.

**Review by Archmonger — APPROVED** [review]
Minor nitpick on CI workflow speed.

Other than that looks good to me.

I'll leave the final review in @Natim 's hands

File: .github/workflows/test.yml
Comment (tari) [reply .github/workflows/test.yml]: Seems like simply removing the limit is most reasonable.

**Comment by Archmonger** [comment]
@Natim Just a reminder that this PR needs merging. Package is currently broken with Django 4+.
I've confirmed the changes look good, but am leaving the final say to you.

**Pull request merged**
